auth/src/main/java/feast/auth/config/CacheConfiguration.java

```diff
@@ -0,0 +1,102 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.auth.config;
+
+import com.google.common.cache.CacheBuilder;
+import feast.auth.utils.AuthUtils;
+import java.lang.reflect.Method;
+import java.util.concurrent.TimeUnit;
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
+import org.springframework.cache.annotation.CachingConfigurer;
+import org.springframework.cache.annotation.EnableCaching;
+import org.springframework.cache.concurrent.ConcurrentMapCache;
+import org.springframework.cache.concurrent.ConcurrentMapCacheManager;
+import org.springframework.cache.interceptor.CacheErrorHandler;
+import org.springframework.cache.interceptor.CacheResolver;
+import org.springframework.cache.interceptor.KeyGenerator;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.core.Authentication;
+
+/** CacheConfiguration class defines Cache settings for HttpAuthorizationProvider class. */
+@Configuration
+@EnableCaching
+@Setter
+@Getter
+public class CacheConfiguration implements CachingConfigurer {
+
+  private static final int CACHE_SIZE = 10000;
+
+  public static int TTL = 60;
+
+  public static final String AUTHORIZATION_CACHE = "authorization";
+
+  @Autowired SecurityProperties secutiryProps;
+
+  @Bean
+  public CacheManager cacheManager() {
+    ConcurrentMapCacheManager cacheManager =
+        new ConcurrentMapCacheManager(AUTHORIZATION_CACHE) {
+
+          @Override
+          protected Cache createConcurrentMapCache(final String name) {
+            return new ConcurrentMapCache(
+                name,
+                CacheBuilder.newBuilder()
+                    .expireAfterWrite(TTL, TimeUnit.SECONDS)
+                    .maximumSize(CACHE_SIZE)
+                    .build()
+                    .asMap(),
+                false);
+          }
+        };
+
+    return cacheManager;
+  }
+
+  @Override
+  public CacheResolver cacheResolver() {
+    // TODO Auto-generated method stub
+    return null;
+  }
+
+  /*
+   * KeyGenerator for HttpAuthorizationProvider.checkAccessToProject() Method.
+   * Key format : checkAccessToProject-<projectId>-<subjectClaim>
+   */
+  @Override
+  public KeyGenerator keyGenerator() {
+    return (Object target, Method method, Object... params) -> {
+      String projectId = (String) params[0];
```

Comment:
Would it be better to try and have a conditional based on the method name?

Reply:
Do you mean use `subClaim` if it is `checkAccessToProject` and default key generator otherwise?. It would be cleaner to write an other key generator or use spring's default key generator if we add other cache in future.

Reply:
Please see my comment above.